R/use_extendr.R

```diff
@@ -195,43 +101,56 @@ extendr_module! {{
   return(invisible(TRUE))
 }
 
-#' Creates example R wrappers for Rust functions.
+#' Write templates from `inst/templates`
+#'
+#' `use_rextendr_template()` is a wrapper around `usethis::use_template()` when
+#' it's available and otherwise implements a simple version of `use_template()`.
+#'
+#' @inheritParams usethis::use_template
+#' @inheritParams use_extendr
 #'
-#' Writes simple R wrappers, without which initial compilation of the
-#' package is impossible. Does not require a compiled library.
-#' Can be used as a fallback.
-#' @param pkg_name The name of the package.
-#' @param outfile Determines where to write wrapper code.
-#' @param path Path from which package root is looked up. Used for pretty message formatting.
-#' @param extra_items Character vector or `NULL` containing additional wrappers.
-#' Should be valid R code.
-#' @param quiet Logical scalar indicating whether the output should be quiet (`TRUE`)
-#'   or verbose (`FALSE`).
 #' @noRd
-make_example_wrappers <- function(pkg_name, outfile, path = ".", extra_items = NULL, quiet = FALSE) {
-  roxcmt <- "#'" # workaround for roxygen parsing bug in raw strings
-
-  wrappers_content <- glue::glue(
-    r"(
-    {roxcmt} @docType package
-    {roxcmt} @usage NULL
-    {roxcmt} @useDynLib {pkg_name}, .registration = TRUE
-    NULL
-    )"
-  )
-
-  if (!is.null(extra_items)) {
-    wrappers_content <- glue::glue_collapse(
-      c(wrappers_content, extra_items),
-      sep = "\n"
+use_rextendr_template <- function(template, save_as = template, data = list(), quiet = getOption("usethis.quiet", FALSE)) {
+  if (is_installed("usethis")) {
+    created <- usethis::use_template(
+      template,
+      save_as = save_as,
+      data = data,
+      open = FALSE,
+      package = "rextendr"
```

Comment:
How will `use_template` behave if we pass `quiet = TRUE` to `use_extendr_template` when the global option `usethis.quiet = FALSE`?
Is it possible that without {usethis} our implementation will be quiet, but **with** {usethis} it will be verbose?
If yes, perhaps we should wrap call to `use_template` in some `with*` method to explicitly set up {usethis} `quiet` option for the duration of its execution?

Reply:
Yes, you're right about that. Good catch! Luckily we can safely assume withr is installed since it's a dep of usethis, so that makes it easy.